### PR TITLE
GH-38652: [MATLAB] Add tests about time precision preservation when converting MATLAB duration to `arrow.array.Time32Array` and `arrow.array.Time64Array`

### LINKED ISSUE #38652: [MATLAB] Add tests about time precision preservation when converting MATLAB duration to arrow time32 and time64
### Describe the enhancement requested

The current tests for MATLAB interface `arrow.array.Time32Array.fromMATLAB` and `arrow.array.Time64Array.fromMATLAB` don't cover the preservation of time precision when `TimeUnit` is specified. The design might be changed in the future. If we add tests to cover the current behavior, the tests will be helpful to evaluate impact in the future.

**An example of current behavior:**
```matlab
>> t = duration(0,0,1.5, 'Format', 'hh:mm:ss.SSS')
t = 
  duration
   00:00:01.500

>> arrow.array.Time32Array.fromMATLAB(t)
ans = 
  Time32Array with 1 element and 0 null values:
    00:00:02
```

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by leihou6116**
### Rationale for this change

The current conversion from MATLAB duration to `arrow.array.Time32Array` and `arrow.array.Time64Array` loses time precision, and there is no test to cover such limitation. It is best practice to have tests cover software design. In addition, such tests will be helpful to evaluate the impact in the future when we improve the design.  

### What changes are included in this PR?  

I mainly added three test cases for each of `arrow.array.Time32Array` and `arrow.array.Time64Array`.

- Updated the basic test case to verify both class and value. In the MATLAB interface tests, we would like to verify the value to make sure there is no precision loss. The basic test case will serve as a test example when people learn to write tests. Updating the basic test case will set a good example for contributors to learn. 
- Test the default value of "TimeUnit".
- Test the functionality of "TimeUnit".

### Are these changes tested?

No software change. The updated test files passed on my local machine.

### Are there any user-facing changes?

No
* Closes: #38652

### CODE REVIEW

File: matlab/test/arrow/array/tTime32Array.m [hunk 1]
```diff
@@ -30,6 +30,14 @@ function Basic(tc)
             times = seconds(1:4);
             array = tc.ArrowArrayConstructorFcn(times);
             tc.verifyInstanceOf(array, "arrow.array.Time32Array");
+            tc.verifyEqual(array.toMATLAB, times');
+        end
+
+        function TimeUnitDefaultValue(tc)
+            % Verify that the default value of "TimeUnit" is "second".
```
Comment (sgilmore10) [inline matlab/test/arrow/array/tTime32Array.m:37]: I think you forgot to check that the `array.Type.TimeUnit` is equal to `Second` based on the comment.

File: matlab/test/arrow/array/tTime64Array.m [hunk 2]
```diff
@@ -30,6 +30,25 @@ function Basic(tc)
             times = seconds(1:4);
             array = tc.ArrowArrayConstructorFcn(times);
             tc.verifyInstanceOf(array, "arrow.array.Time64Array");
+            tc.verifyEqual(array.toMATLAB, times');
+        end
+
+        function TimeUnitDefaultValue(tc)
+            % Verify that the default value of "TimeUnit" is "Microsecond".
```
Comment (sgilmore10) [inline matlab/test/arrow/array/tTime64Array.m:37]: I think you should check the `array.Type.TimeUnit` is `Microsecond`

File: matlab/test/arrow/array/tTime64Array.m [hunk 3]
```diff
@@ -290,6 +310,66 @@ function TestIsEqualFalseTimeUnitMistmatch(tc)
...
+                                    1.123456789])',...
+                          'AbsTol',seconds(1e-15));
+        end
+
+        function TimeUnitIsReadOnly(tc)
+            % Verify that arrowArray.Type.TimeUnit cannot be changed.
+
+            matlabTimes = seconds([1.000001, 2.999999, 0.0002004]);
+            arrowArray = tc.ArrowArrayConstructorFcn(matlabTimes);
+            tc.verifyError(@()setTimeUnit(arrowArray),'MATLAB:class:SetProhibited');
```
Comment (kevingurney) [inline matlab/test/arrow/array/tTime64Array.m:367]: You can use `setfield` here like:

https://github.com/apache/arrow/blob/bff5fb95ac987354ecd7d69466e11b9f33a67d08/matlab/test/arrow/tabular/tRecordBatch.m#L260

**Review by kevingurney — CHANGES_REQUESTED** [review]
Thank you for adding these tests, @leihou6116!

**Comment by kevingurney** [comment]
+1

**Review by kevingurney — APPROVED** [review]

**Pull request merged**